File: week08/5/prob1.cpp
// you can use includes, for example:
// #include <algorithm>
#include <vector>
using namespace std;

// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;

int solution(int K, vector<int> &A) {
    // Implement your solution here
    int ropesNum = 0;
    
    int currentSum = 0;
    for (int len : A) {
        // not enough length
        if (currentSum + len < K) currentSum += len;
        else {
            currentSum = 0;
            ropesNum += 1;
        }
    }

    return ropesNum;
}

File: week08/2/Prob1.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(K, A):
    # Implement your solution here
    sum = 0
    count = 0
    for i in A:
        sum += i
        if sum >= K:
            count += 1
            sum = 0
    return count